helm/chart-operator/templates/_helpers.tpl

```diff
@@ -19,7 +19,7 @@ Common labels
 {{- define "chart-operator.labels" -}}
 {{ include "chart-operator.selectorLabels" . }}
 app: {{ include "chart-operator.name" . | quote }}
-app.giantswarm.io/branch: {{ .Values.project.branch | replace "#" "-" | replace "/" "-" | trunc 63 | quote }}
+app.giantswarm.io/branch: {{ .Values.project.branch | replace "#" "-" | replace "/" "-" | replace "." "-" | trunc 63 | trimSuffix "-" | quote }}
```

Comment:
This fix is needed. Otherwise the branch label is invalid and the integration tests fail.

This change is already in app-operator.